components/brave_wallet_ui/page/wallet_page.tsx

```diff
@@ -5,34 +5,139 @@
+import BuySendSwap from '../components/buy-send-swap'
 import Theme from 'brave-ui/theme/brave-default'
 import DarkTheme from 'brave-ui/theme/brave-dark'
 import BraveCoreThemeProvider from '../../common/BraveCoreThemeProvider'
+import walletDarkTheme from '../theme/wallet-dark'
+import walletLightTheme from '../theme/wallet-light'
+
+type Props = {
+  page: WalletPageReducerState
+  actions: typeof WalletPageActions
+}
+
+function Page (props: Props) {
+  const [view, setView] = React.useState<NavTypes>('crypto')
+  const [linkedAccounts] = React.useState<NavObjectType[]>(LinkedAccountsOptions)
+
+  // In the future these will be actual paths
+  // for example wallet/rewards
+  const navigateTo = (path: NavTypes) => {
+    setView(path)
+  }
+
+  return (
+    <WalletPageLayout>
+        <SideNav
+          navList={NavOptions}
+          staticList={StaticOptions}
+          selectedButton={view}
+          onSubmit={navigateTo}
+          linkedAccountsList={linkedAccounts}
+        />
+        <WalletSubViewLayout>
+          {view === 'crypto' ? (
+            <CryptoView />
+          ) : (
+            <div style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+              <h2>{view} view</h2>
+            </div>
+          )}
+
+        </WalletSubViewLayout>
+        <WalletWidgetStandIn>
+          <BuySendSwap />
+        </WalletWidgetStandIn>
+      </WalletPageLayout>
+  )
+}
+
+function mapStateToProps (state: State): Partial<Props> {
+  return {
+    page: state.walletPageReducer
+  }
+}
+
+function mapDispatchToProps (dispatch: Dispatch): Partial<Props> {
+  return {
+    actions: bindActionCreators(WalletPageActions, store.dispatch.bind(store))
+  }
+}
+
+const PageWithState = connect(mapStateToProps, mapDispatchToProps)(Page)
+
+function App () {
+  const [initialThemeType, setInitialThemeType] = React.useState<chrome.braveTheme.ThemeType>()
+  React.useEffect(() => {
+    chrome.braveTheme.getBraveThemeType(setInitialThemeType)
+  }, [])
+  return (
+    <Provider store={store}>
+      {initialThemeType &&
+      <BraveCoreThemeProvider
+        initialThemeType={initialThemeType}
+        dark={walletDarkTheme}
+        light={walletLightTheme}
+      >
+        <PageWithState
+        />
+      </BraveCoreThemeProvider>
+      }
+    </Provider>
+  )
+}
 
 function initialize () {
+  chrome.braveWallet.isNativeWalletEnabled((enabled: boolean) => {
+    if (enabled) {
+      store.dispatch(WalletPageActions.initialize())
+      render(<App />, document.getElementById('root'))
+    } else {
+      initializeOldWallet()
+    }
+  })
+}
+
+function initializeOldWallet () {
   chrome.braveWallet.shouldPromptForSetup((prompt: boolean) => {
     if (!prompt) {
       chrome.braveWallet.loadUI(() => {
         window.location.href = 'chrome://wallet'
       })
       return
     }
-
-    renderWebUIView()
+    renderOldWebUIView()
   })
 }
 
-function renderWebUIView () {
+function renderOldWebUIView () {
```

Comment:
I'll do this splitting in a separate PR per Slack. Mainly the motivation for why I did this was because the old page is just a container which redirects to an extension mapping.  And I thought it might be better to decrease the size of common code like React that's used in both. 

I agree though that it'd be better to handle the common code via a separate common bundle as I think we have a lot of bundles today with a lot of duplication of deps.

I posted this issue to track the work of splitting out the old / new page though for now: https://github.com/brave/brave-browser/issues/15756